src/recap

```diff
@@ -146,492 +134,440 @@ print_usage() {
 
 # create output file
 create_output_file() {
-OUTPUT_FILE=$1
-
-if [ -d $OUTPUT_FILE ]
-then
-	echo "Target file already exists: $OUTPUT_FILE"
-	exit
-else
-	#print the data to the output file
-	echo "$DATE" > $OUTPUT_FILE
-fi
+  local OUTPUT_FILE=$1
+  if [[ -d "${OUTPUT_FILE}" ]]; then
+    echo "Target file already exists: ${OUTPUT_FILE}"
+    exit
+  else
+    #print the data to the output file
+    echo "${DATE}" > "${OUTPUT_FILE}"
+  fi
 }
 
 # check to see if the output directory exists
 check_output_file() {
-OUTPUT_FILE=$1
-
-if [ ! -f $OUTPUT_FILE ]
-then
-	echo "The output file does not exist: $OUTPUT_FILE"
-	exit
-fi
+  local OUTPUT_FILE=$1
+  if [[ ! -f "${OUTPUT_FILE}" ]]; then
+    echo "The output file does not exist: ${OUTPUT_FILE}"
+    exit
+  fi
 }
 
-# print output of "ps auxfww" to the PS_FILE
+# print output of "ps auxfww" to the specified file
 print_ps() {
-	ps $OPTS_PS >> $PS_FILE
+  local LOGFILE=$1
+  ps ${OPTS_PS} >> "${LOGFILE}"
 }
 
-# print output of "pstree" to PSTREE_FILE
+# print output of "pstree" to specified file
 print_pstree() {
-	pstree $OPTS_PSTREE >> $PSTREE_FILE
+  local LOGFILE=$1
+  pstree ${OPTS_PSTREE} >> "${LOGFILE}"
 }
 
 # print a blank line to the specified file
 print_blankline() {
-	echo " " >> $1
+  local LOGFILE=$1
+  echo " " >> "${LOGFILE}"
 }
 
-# print the output of "uptime" to the resources output file
+# print the output of "uptime" to the specified file
 print_uptime() {
-	echo "UPTIME report" >> $RESOURCES_FILE
-	uptime >> $RESOURCES_FILE
+  local LOGFILE=$1
+  echo "UPTIME report" >> "${LOGFILE}"
+  uptime >> "${LOGFILE}"
 }
 
-# print the output of "free" to the resources file
+# print the output of "free" to the specified file
 print_free() {
-	echo "FREE report" >> $RESOURCES_FILE
-	free $OPTS_FREE >> $RESOURCES_FILE
+  local LOGFILE=$1
+  echo "FREE report" >> "${LOGFILE}"
+  free ${OPTS_FREE} >> "${LOGFILE}"
 }
 
-# print the output of "vmstat" to the resources file
+# print the output of "vmstat" to the specified file
 print_vmstat() {
-	echo "VMSTAT report" >> $RESOURCES_FILE
-	vmstat $OPTS_VMSTAT >> $RESOURCES_FILE
+  local LOGFILE=$1
+  echo "VMSTAT report" >> "${LOGFILE}"
+  vmstat ${OPTS_VMSTAT} >> "${LOGFILE}"
 }
 
-# print the output of "iostat" to the resources file
+# print the output of "iostat" to the specified file
 print_iostat() {
-	echo "IOSTAT report" >> $RESOURCES_FILE
-	iostat $OPTS_IOSTAT >> $RESOURCES_FILE
+  local LOGFILE=$1
+  echo "IOSTAT report" >> "${LOGFILE}"
+  iostat ${OPTS_IOSTAT} >> "${LOGFILE}"
 }
 
-# print the output of "iotop" to the resources file
+# print the output of "iotop" to the specified file
 print_iotop() {
-	which iotop >/dev/null 2>&1 || return 1
-	echo "IOTOP report" >> $RESOURCES_FILE
-	iotop $OPTS_IOTOP >> $RESOURCES_FILE
+  local LOGFILE=$1
+  which iotop >/dev/null 2>&1 || return 1
```

Comment:
to me seems like a remainder, I rather remove this or add to every single tool used,  ideally recap would be installed from a package(rpm or deb) in which case the dependency should be solved and  recap code should not double check that, for instance I've the list of package dependencies already sorted out in here https://github.com/tonyskapunk/recap/blob/recap_v1/util/packaging/rpm/recap.spec#L11 there is not a PR of this yet until this PR #78 and recaplog PR #72 are merged in master.  DEB packages are not in this repo, that's maintained outside the code, tagging @thebwt  as he is the packager for DEBs to see if the deb config is publicly available.  Finally on AUR I'm the maintainer of the package(currently recap-git) and is a dependency in there: https://aur.archlinux.org/packages/recap-git/